Documentation/GettingStarted.md

```diff
@@ -1,104 +1,861 @@
-
-// To pull values out of rx variable `c`, subscribe to values from  `c`.
-// `subscribeNext` means subscribe to next (fresh) values of variable `c`.
-// That also includes the inital value "3 is positive".
-c >- subscribeNext { println($0) }          // prints: "3 is positive"
-
-// Now let's increase the value of `a`
-// a = 4 is in RxSwift
-a.next(4)                                   // prints: 6 is positive
-// Sum of latest values is now `4 + 2`, `6` is >= 0, map operator
-// produces "6 is positive" and that result is "assigned" to `c`.
-// Since the value of `c` changed, `{ println($0) }` will get called, 
-// and "6 is positive" is printed.
-
-// Now let's change the value of `b`
-// b = -8 is in RxSwift
-b.next(-8)                                  // doesn't print anything
-// Sum of latest values is `4 + (-8)`, `-4` is not >= 0, map doesn't 
-// get executed.
-// That means that `c` still contains "6 is positive" and that's correct.
-// Since `c` hasn't been updated, that means next value hasn't been produced,
-// and `{ println($0) }` won't be called.
+let counter = myInterval(0.1)
+    >- replay(1)
+    >- refCount
 
-// ...
+println("Started ----")
+
+let subscription1 = counter
+    >- subscribeNext { n in
+       println("First \(n)")
+    }
+let subscription2 = counter
+    >- subscribeNext { n in
+       println("Second \(n)")
+    }
+
+NSThread.sleepForTimeInterval(0.5)
+
+subscription1.dispose()
+
+NSThread.sleepForTimeInterval(0.5)
+
+subscription2.dispose()
+
+println("Ended ----")
+```
+
+this will print
+
+```
+Started ----
+Subscribed
+First 0
+Second 0
+First 1
+Second 1
+First 2
+Second 2
+First 3
+Second 3
+First 4
+Second 4
+First 5
+Second 5
+Second 6
+Second 7
+Second 8
+Second 9
+Disposed
+Ended ----
+```
+
+Notice how now there is only one `Subscribed` and `Disposed` event.
+
+This pattern of sharing subscriptions is so common in UI layer that it has it's own operator. Instead of writing `>- replay(1) >- refCount`, you can just write `>- variable`.
+
+Behavior for URL observables is equivalent.
+
+This is how HTTP requests are wrapped in Rx. It's pretty much the same pattern like the `interval` operator.
+
+```swift
+extension NSURLSession {
+    public func rx_response(request: NSURLRequest) -> Observable<(NSData!, NSURLResponse!)> {
+        return create { observer in
+            let task = self.dataTaskWithRequest(request) { (data, response, error) in
+                if data == nil || response == nil {
+                    sendError(observer, error ?? UnknownError)
+                }
+                else {
+                    sendNext(observer, (data, response))
+                    sendCompleted(observer)
+                }
+            }
+
+            task.resume()
+
+            return AnonymousDisposable {
+                task.cancel()
+            }
+        }
+    }
+}
+```
+
+## Operators
+
+There are numerous operators implemented in RxSwift. Complete list can be found [here](API.md).
```

Comment:
> There are numerous operators implemented in RxSwift. Complete list can be found [here](API.md).

There are numerous operators implemented in RxSwift. The complete list can be found [here](API.md).